File: examples/python/datasets_example.py
import qoery

configuration = qoery.Configuration()
configuration.api_key['ApiKeyAuth'] = 'YOUR_API_KEY'

with qoery.ApiClient(configuration) as api_client:
    api = qoery.DatasetsApi(api_client)
    
    # Start job
    job = api.start_dataset_job(search="climate change")
    print(f"Job ID: {job.job_id}")
    
    # Get status
    status = api.get_dataset_job_status(job.job_id)
    print(f"Status: {status.status}")
    
    # Download CSV
    csv = api.download_dataset_csv(job.job_id)
    print(f"CSV: {len(csv)} bytes")

File: examples/js/datasets_example.js
const Qoery = require('qoery');

const client = Qoery.ApiClient.instance;
client.authentications['ApiKeyAuth'].apiKey = 'YOUR_API_KEY';

const api = new Qoery.DatasetsApi();

// Start job
api.startDatasetJob({ search: 'climate change' }, (error, job) => {
  if (error) throw error;
  console.log(`Job ID: ${job.job_id}`);
  
  // Get status
  api.getDatasetJobStatus(job.job_id, (error, status) => {
    if (error) throw error;
    console.log(`Status: ${status.status}`);
    
    // Download CSV
    api.downloadDatasetCsv(job.job_id, (error, csv) => {
      if (error) throw error;
      console.log(`CSV: ${csv.size || csv.length} bytes`);
    });
  });
});
